libs/datamodel/core/src/configuration/datasource.rs

```diff
@@ -22,6 +23,25 @@ impl Datasource {
     }
 }
 
+pub trait DatasourcePreviewFeatures {
```

Comment:
You can just directly implement the `PreviewFeatures` trait from `generator.rs`. Probably it would be good to extract the trait into its own file. Then it's clearer that both `Generator` and `Datasource` can both implement it.